editors/vscode/src/main.ts

```diff
@@ -243,15 +244,27 @@ async function getWorkspaceDependency(
 	outputChannel: OutputChannel,
 ): Promise<string | undefined> {
 	for (const workspaceFolder of workspace.workspaceFolders) {
-		const path = Uri.joinPath(
-			workspaceFolder.uri,
-			"node_modules",
-			".bin",
-			"biome",
+		// To resolve the @biomejs/cli-*, which is a transitive dependency of the
+		// @biomejs/biome package, we need to create a custom require function that
+		// is scoped to @biomejs/biome. This allows us to reliably resolve the
+		// package regardless of the package manager used by the user.
+		const requireFromBiome = createRequire(
+			require.resolve("@biomejs/biome/package.json", {
```

Comment:
In the past, I've had issues with `require.resolve`. Not just with `rome` but also in Astro when working on requiring some package. `require.resolve` has some caching strategy that is... weird.

Can't we use https://www.npmjs.com/package/resolve ? It worked in `rome` and it works in `astro`

Reply:
I'll git it a try.

Reply:
I don't think that I can leverage createRequire when using this package, which defeats the purpose. What were the issues encountered when using resolve.require and why were they a concern in the VS Code extension ?

Reply:
No worries, let's try `resolve.require`. 

> What were the issues encountered when using resolve.require and why were they a concern in the VS Code extension ?

I don't really remember the reason, but here the PR when it was implemented: https://github.com/rome/tools/pull/3717

But if you your fix rework too, let's just go with that!